### PR TITLE
Nodejs week1

### PR DESCRIPTION
**Pull request by GizemSavci**

### CODE REVIEW

File: src/backend/app.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import mealsRouter from "./api/meals.js";
 import cors from "cors";
 import { fileURLToPath } from 'url';
 import { dirname } from 'path';
-
+import knex from 'knex';
```
Comment (kiko-bkr) [inline src/backend/app.js:7]: Here you are importing the `knex` function from the `knex` library.  You have already done this in the `database.js` file, where you used the function to connect to your MySQL instance. Thus you need to import the actual connection object `myKnex` from that file. This line should be:
```javascript
import knex from "./database.js";
```

File: src/backend/app.js [hunk 2]
```diff
@@ -27,6 +27,86 @@ app.use(cors());
...
+app.get("/first-meal", async (req, res) => {
+  try {
+    const firstMeal = await knex("meal")
+      .select()
+      .orderBy("id")
+      .first();
+    res.status(200).json(firstMeal)
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Error");
```
Comment (kiko-bkr) [inline src/backend/app.js:92]: Nicely written query! This route requests one singular meal, and if that one is not found, the status `404` should be returned with an appropriate message. Same thing applies for `/last-meal`.

**Review by kiko-bkr — CHANGES_REQUESTED** [review]
Great job overall! 🥇 There are only a few considerations to be changed. Make sure to pay attention to the assignment specifications. 😉

File: src/backend/app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (GizemSavci) [reply src/backend/app.js:7]: thank you, implemented😊

File: src/backend/app.js [hunk 3]
```diff
@@ -27,6 +27,86 @@ app.use(cors());
...
+app.get("/future-meals", async (req, res) => {
+  try {
+    const futureMeals = await knex("meal")
+      .select()
+      .where("meal_time", ">", new Date());
+    if (futureMeals.length === 0) {
+      res.status(404).send("No meals found");
+    } else {
+      res.status(200).json(futureMeals);
+    }
```
[the remark this answers is not in the document]
Comment (GizemSavci) [reply src/backend/app.js:45]: So, should it be `    if (futureMeals.length >= 0) {
      res.status(200).json(futureMeals);
    } else {
      res.status(404).send("No meals found");
    }
  } catch (error) {
    console.error(error);
    res.status(500).send("Internal Server Error");
  }
});

[the remark this answers is not in the document]
Comment (GizemSavci) [reply src/backend/app.js:45]: Or, should it be `    const futureMeals = await knex("meal")
      .select()
      .where("meal_time", ">", new Date());

      res.status(200).json(futureMeals);`

File: src/backend/app.js (hunk 2 above)
Comment (GizemSavci) [reply src/backend/app.js:92]: So it should be: `    if (firstMeal.length === 0){
      res.status(404).send("No meal found");
    }else {
    res.status(200).json(firstMeal);
    }`

Comment (kiko-bkr) [reply src/backend/app.js:89]: Yes, that would definitely work!

File: src/backend/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kiko-bkr) [reply src/backend/app.js]: Precisely! For future reference, [this is how](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) to insert multi-line code in these comments.

**Review by kiko-bkr — APPROVED** [review]
Corrected successfully! 👍

File: src/backend/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (GizemSavci) [reply src/backend/app.js]: Thank you😊

File: src/backend/app.js (hunk 2 above)
Comment (GizemSavci) [reply src/backend/app.js:89]: thank you😊

**Pull request merged**